sphinxcontrib/katex.py

```diff
@@ -74,31 +75,19 @@ def get_latex(node):
         return node.astext()  # for Sphinx >= 1.8.0
 
 
+def run_katex(latex):
+    p = subprocess.Popen(('katex', ), stdout=subprocess.PIPE, stdin=subprocess.PIPE)
+    stdout, stderr = p.communicate(latex.encode('utf-8'))
+    return stdout.decode('utf-8')
+
+
 def html_visit_math(self, node):
-    self.body.append(self.starttag(node, 'span', '', CLASS='math'))
-    self.body.append(self.builder.config.katex_inline[0] +
-                     self.encode(get_latex(node)) +
-                     self.builder.config.katex_inline[1] + '</span>')
+    self.body.append(run_katex(get_latex(node)))
     raise nodes.SkipNode
 
 
 def html_visit_displaymath(self, node):
-    self.body.append(self.starttag(node, 'div', CLASS='math'))
-    if node['nowrap']:
```

Comment:
I'm not exactly sure what to do with nowrap and number

Reply:
There is one little change with your current solution:
Before if `nowrap` was given, no equation number was added. Now, the equation number is always added if `node['number']` is present or not. But I cannot even say which behavior is better ;)